libcaf_net/src/application.cpp

```diff
@@ -47,6 +48,81 @@ application::application(proxy_registry& proxies) : proxies_(proxies) {
   // nop
 }
 
+error application::write_message(
+  packet_writer& writer, std::unique_ptr<endpoint_manager_queue::message> ptr) {
+  CAF_ASSERT(ptr != nullptr);
+  CAF_ASSERT(ptr->msg != nullptr);
+  CAF_LOG_TRACE(CAF_ARG2("content", ptr->msg->content()));
+  auto payload_elem_buf = writer.next_buffer();
```

Comment:
```suggestion
  auto payload_prefix = writer.next_buffer();
```

Probably a better name. I don't have any intuition what `elem_buf` should mean.